File: server/routes/content.js
const express = require("express");
const sqlite3 = require("sqlite3").verbose();

const router = express.Router();
const db = new sqlite3.Database("../database.sqlite");

// Get all content
router.get("/", (req, res) => {
  db.all("SELECT * FROM content", [], (err, rows) => {
    if (err) return res.status(500).send("Database error");
    res.json(rows);
  });
});

// Update content
router.put("/:id", (req, res) => {
  const { id } = req.params;
  const { section_name, content } = req.body;

  db.run(
    "UPDATE content SET section_name = ?, content = ? WHERE id = ?",
    [section_name, content, id],
    function (err) {
      if (err) return res.status(500).send("Database error");
      res.json({ updated: this.changes });
    }
  );
});

module.exports = router;

File: server/routes/content.py
from flask import Blueprint, request, jsonify
import sqlite3

content_bp = Blueprint('content', __name__)

def get_db_connection():
    conn = sqlite3.connect('database.sqlite')
    conn.row_factory = sqlite3.Row
    return conn

@content_bp.route('/', methods=['GET'])
def get_content():
    conn = get_db_connection()
    content = conn.execute('SELECT * FROM content').fetchall()
    conn.close()

    return jsonify([dict(row) for row in content])

@content_bp.route('/<int:id>', methods=['PUT'])
def update_content(id):
    data = request.get_json()
    section_name = data.get('section_name')
    content = data.get('content')

    conn = get_db_connection()
    conn.execute('UPDATE content SET section_name = ?, content = ? WHERE id = ?', (section_name, content, id))
    conn.commit()
    conn.close()

    return jsonify({'message': 'Content updated successfully'})